src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,232 @@
+                  </Button>
+              </>
+          }
+        >
+            <Stack>
+                {createError &&
+                <Alert
+                  className="file-editor-alert"
+                  variant="danger"
+                  title={createError}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup
+                      label={_("File name")}
+                      fieldId="file-name"
+                    >
+                        <TextInput
+                          autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                          id="file-name"
+                          isRequired
+                          value={filename}
+                          onChange={(_event, value) => {
+                              setFileNameError(checkFilename(value, cwdInfo?.entries || {}, undefined));
+                              setFilename(value);
+                          }}
+                        />
+                        <FormHelper fieldId="file-name" helperTextInvalid={filenameError} />
+                    </FormGroup>
+                    <TextArea
+                      id="editor-text-area"
+                      className="file-editor"
+                      value={initialText}
+                      onChange={(_ev, content) => setInitialText(content)}
+                    />
+                    {candidates.length > 0 &&
+                        <FormGroup fieldId="create-file-owner" label={_("Owner")}>
+                            <FormSelect
+                              id="create-file-owner"
+                              value={owner}
+                              onChange={(_ev, val) => setOwner(val)}
+                            >
+                                {candidates.map(owner =>
+                                    <FormSelectOption
+                                      key={owner}
+                                      value={owner}
+                                      label={owner}
+                                    />)}
+                            </FormSelect>
+                        </FormGroup>}
+                </Form>
+            </Stack>
+        </Modal>
+    );
+};
+
+export async function show_create_file_dialog(
+    dialogs: Dialogs,
+    path: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void
+) {
+    if (!superuser.allowed) {
+        try {
+            await cockpit.spawn(["test", "-w", path]);
+        } catch {
+            addAlert(_("Cannot create file in current directory"), AlertVariant.warning, "create-file",
+                     _("Permission denied"));
+            return;
+        }
+    }
+
+    // A cancelled dialog means no file is created so don't open the editor
+    try {
+        const filename = await dialogs.run(CreateFileModal, { path });
+        edit_file(dialogs, filename);
+    } catch {
+    }
```

Comment:
Is cancel really the only way how CreateFileModal could fail? No undefined variables, failures during write, etc? It would be good to actually check the error that it really is a cancel and otherwise cockpit.error(). Or do cancel via a `null` resolve instead of a reject, i.e. not as an error.